rfcs/0001-extended-lookup-tables.md

```diff
@@ -0,0 +1,530 @@
+polynomial `FixedAndRuntimeTable`, giving its evaluations and openings as part
+of the proof, and asserting that each entry matches the above.
+
+The `RuntimeTableSelector` polynomial will be fixed at constraint system time,
+and we can avoid opening it. However, we must provide openings for
+`RuntimeTable(i)`.
+
+The `joint_combiner` should be sampled using randomness that depends on the
+individual `RuntimeTable(i)`s, to ensure that the runtime values cannot be
+chosen with prior knowledge `joint_combiner`.
+
+### Representing the side-loaded tables
+
+The side-loaded tables can be considered as additional 'masks' that we apply to
+the combined fixed and runtime table, to introduce this side-loaded data.
+
+Again, we make the simplifying assumption that an entry from a side-loaded
+table has exactly 1 'fixed' entry, which can be used for indexing (or set to
+`0` if unused). We also assume that at most 1 side-loaded table contributes to
+each entry.
+
+In order to compute the lookup multiset inclusion argument, we also need the
+combined table of all 3 table kinds, which we will call `LookupTable`. We can
+calculate the contribution of the side-loaded table for each lookup as
+```
+LookupTable - FixedAndRuntimeTable
+```
+
+We will represent the polynomials for the side-loaded tables as
+`SideLoadedTable(i, j)`, where `i` identifies the table, and `j` identifies the
+column within that table. We also include polynomials `SideLoadedCombined(i)`
+in the proof, along with their openings, where
+```
+SideLoadedCombined(i)
+  = sum[j=1, ?, SideLoadedTable(i, j) * joint_combiner^(j-1)]
+```
+and check for consistency of the evaluations and openings when verifying the
+proof.
+
+The `SideLoadedTable` polynomials are included in the proof, but we do so 'as
+if' they had been part of the verification index (ie. without any
+proof-specific openings), so that they may be reused across multiple different
+proofs without modification.
+
+The `joint_combiner` should be sampled using randomness that depends on the
+individual `SideLoadedTable(i,j)`s, to ensure that the side-loaded values
+cannot be chosen with prior knowledge `joint_combiner`.
+
+*Note: it may be useful for the verifier to pass the `SideLoadedTable(i,j)`s as
+public inputs to the proof, so that the circuit may assert that the table it
+had side-loaded was signed or otherwise correlated with a cryptographic
+commitment that it had access to. The details of this are deliberately elided
+in this RFC, since the principle and implementation are both fairly
+straightforward.*
+
+### Position-independence of side-loaded tables
+
+We want to ensure that side-loaded tables are maximally re-usable across
+different proofs, even where the 'position' of the same side-loaded table in
+the combined table may vary between proofs. For example, we could consider a
+'folding' circuit `C` which takes 2 side-loaded tables `t_in` and `t_out`,
+where we prove
```

Comment:
the phrasing is very difficult to follow, I'm not sure if I understood this correctly but does `C(t_0, t_1)` check that `t_0` is in `t_in` and `t_1` is in `t_out`?